### PR TITLE
feat(sidebar): Show node owner in metadata subline 

### LINKED ISSUE #46178: [App Files] Can't see owner's name if a file/folder is reshared twice
# Reproduction
**Reproducible ?**
- [x] Yes.
- [ ] No.

**Regularity ?**  
- [x] Always.
- [ ] Sometimes.
- [ ] Unable to reproduce.

**Regression ?**
- [ ] Yes.
- [ ] No.
- [x] New Feature.
 
**Steps to reproduce ?**

1. Go to web app via a main account (for the test: admin)
2. Go to app files
3. Do a collab sharing on a file/folder to a 2nd user (for the test: John testing)
4. Connect on the shared user on web app
5. Do a collab sharing to a 3rd user on the shared file/folder
6. Check on the 3rd user who shared the file/folder

**Obtained results:** Shows that the file/folder is shared by the 2nd user but we can only see who is the file/folder's owner in the bubble in the file view

Image

# Environment and versions
| 26.0.4.3  |  

Browser used:
- [x] Chrome Windows
- [x] Firefox Windows

### PR DESCRIPTION
**Pull request by nfebe**
## Summary

Show user bubble to indicate owner of file in sidebar overview (this should not change irrespective of share status)... 

Resolves: https://github.com/nextcloud/server/issues/46178

## Screenshots
| Timeline | State |
|----------| -------------| 
| Before|metadata-before|
| Now (owned by you) |metadata-owned-by-you |
| Now (owned by other) |metadata-owned-by-other|

### TODO
- [x] Finish wip
- [x] Screenshots

### CODE REVIEW

File: apps/files/src/views/Sidebar.vue [hunk 1]
```diff
@@ -549,6 +574,12 @@ export default {
 		handleWindowResize() {
 			this.hasLowHeight = document.documentElement.clientHeight < 1024
 		},
+		sharedMultipleTimes() {
+			// Mixed share types
+			if (Array.isArray(this.node.attributes?.['share-types']) && this.node.attributes?.['share-types'].length > 1) {
+				return t('files', 'Shared multiple times with different people')
+			}
+		},
```
Comment (Fenn-CS) [inline apps/files/src/views/Sidebar.vue:582]: Probably useful to provide this information somewhere in the side?

- Maybe  a property to check if it has been shared multiple times
- Another property for the actual text.

**Comment by Fenn-CS** [comment]
### Show original file owner in side bar

Now that some logic to show the owner is in place with a simple "draft" to show this information with the userbubble near other folder metada can @nextcloud/designers chime in?


show-admin

**Comment by nimishavijay** [comment]
Nice! Good idea using the user bubble component. I would suggest we include only the username in the user bubble, and to separate the different metadata we could use a middle dot `·`. So it would look like

```
11.2 MB · 3 weeks ago · Owned by (👱🏼 Alex Doe)
```

Also the user bubble component should be center aligned with the rest of the text :)

**Review by sorbaugh — APPROVED** [review]

File: apps/files/src/views/Sidebar.vue [hunk 2]
```diff
@@ -17,14 +17,20 @@
...
+			<div class="sidebar__subname">
+				<NcIconSvgWrapper v-if="fileInfo.isFavourited"
+					:path="mdiStar"
+					:name="t('files', 'Favorite')"
+					inline />
+				<span>{{ size }}</span>
+				<span class="sidebar__subname-separator">.</span>
+				<NcDateTime :timestamp="fileInfo.mtime" />
+				<span class="sidebar__subname-separator">.</span>
+				<span>{{ t('files', 'Owned by') }}</span>
```
Comment (nickvergessen) [inline apps/files/src/views/Sidebar.vue:29]: string concatenation is bad for some languages, see https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#improving-your-translations

File: apps/files/src/views/Sidebar.vue [hunk 3]
```diff
@@ -288,6 +295,25 @@ export default {
...
+		},
+		currentUserIsOwner() {
+			return this.ownerId === this.currentUser.uid
+		},
+		nodeOwnerLabel() {
+			if (this.currentUserIsOwner) {
+				return t('files', 'you')
+			}
+			const ownerDisplayName = this.node?.attributes?.['owner-display-name']
+			return ownerDisplayName
```
Comment (nickvergessen) [inline apps/files/src/views/Sidebar.vue:309]: can you test with external storage and groupfolders? I think you end up with an "Owned by" without a name at the moment?

Comment (Fenn-CS) [reply apps/files/src/views/Sidebar.vue:309]: Tested with group folders and it works fine and ti should because the metadata about ownership which is an attribute is "calculated" and sent by the backend irrespective of source or said in another way even in case of external storage, the files and related tables would still hold information about the object but the actual object would come from wherever?

Screenshot from 2024-09-03 11-44-04


Shields

- If the attribute is not available it would show, "Owned by unknown" for now

File: apps/files/src/views/Sidebar.vue (hunk 1 above)
Comment (Fenn-CS) [reply apps/files/src/views/Sidebar.vue:583]: Used to be a feature in previous sharing. cc @jancborchardt

File: apps/files/src/views/Sidebar.vue [hunk 4]
```diff
@@ -17,14 +17,20 @@
...
-			<NcDateTime :timestamp="fileInfo.mtime" />
+			<div class="sidebar__subname">
+				<NcIconSvgWrapper v-if="fileInfo.isFavourited"
+					:path="mdiStar"
+					:name="t('files', 'Favorite')"
+					inline />
+				<span>{{ size }}</span>
+				<span class="sidebar__subname-separator">.</span>
+				<NcDateTime :timestamp="fileInfo.mtime" />
+				<span class="sidebar__subname-separator">.</span>
```
Comment (susnux) [inline apps/files/src/views/Sidebar.vue:28]: Should probably use the same separator as versions

```suggestion
				<span class="sidebar__subname-separator">•</span>
				<NcDateTime :timestamp="fileInfo.mtime" />
				<span class="sidebar__subname-separator">•</span>
```

File: apps/files/src/views/Sidebar.vue [hunk 5]
```diff
@@ -140,12 +145,14 @@ export default {
 	data() {
 		return {
 			// reactive state
+			currentUser: getCurrentUser(),
```
Comment (susnux) [inline apps/files/src/views/Sidebar.vue:148]: Why this change? This is non reactive (the user will not appear or disappear) and should not made reactive (which happens when passing to the data).

And you can access it exactly the same in the computed below (`this.currentUser`)

File: apps/files/src/views/Sidebar.vue (hunk 2 above)
Comment (jancborchardt) [reply apps/files/src/views/Sidebar.vue]: @Fenn-CS if the string concat is not proper, let’s instead do it simpler like `Owner: (avatarbubble)`
Also, make sure to use [middle dots](https://en.wikipedia.org/wiki/Interpunct) to separate the size, date and ownership info: `·`, not regular dots. :)

Comment (Fenn-CS) [reply apps/files/src/views/Sidebar.vue]: ## Screenshots

| Owned by another | Owned by current user |
| --------------------------- | ---------------------------------- |
| Screenshot from 2024-09-17 15-36-27 | Screenshot from 2024-09-17 15-35-26 |

File: apps/files/src/views/Sidebar.vue [hunk 6]
```diff
@@ -460,6 +486,7 @@ export default {
 				this.fileInfo = await FileInfo(this.davPath)
 				// adding this as fallback because other apps expect it
 				this.fileInfo.dir = this.file.split('/').slice(0, -1).join('/')
+				this.node = await fetchNode({ path: (this.fileInfo.path + '/' + this.fileInfo.name).replace('//', '/') })
```
Comment (artonge) [inline apps/files/src/views/Sidebar.vue:489]: Aren't we fetching the file info already somewhere?

File: apps/files/src/views/Sidebar.vue [hunk 7]
```diff
@@ -123,15 +131,14 @@ export default {
 		NcIconSvgWrapper,
 		SidebarTab,
 		SystemTags,
+		NcUserBubble,
 	},
 
 	setup() {
 		const currentUser = getCurrentUser()
-
```
Comment (artonge) [inline apps/files/src/views/Sidebar.vue:130]: Unnecessary change. Same for other whitespaces.

File: apps/files/src/views/Sidebar.vue (hunk 6 above)
Comment (Fenn-CS) [reply apps/files/src/views/Sidebar.vue:493]: I doubt in the same way we need it here, @skjnldsv is there somewhere else I can pick this up?

**Review by sorbaugh — APPROVED** [review]

File: apps/files/src/views/Sidebar.vue (hunk 6 above)
Comment (susnux) [reply apps/files/src/views/Sidebar.vue:493]: As the sidebar is part of the files app, you can use the files store to retirieve the current node without needing to re-fetch it again

Comment (Fenn-CS) [reply apps/files/src/views/Sidebar.vue]: Using the fileStore as at now breaks the sidebar. With undefined errors.... Looks like the sidebar is trying to reference the fileStore before it is ready.

File: apps/files/src/views/Sidebar.vue [hunk 8]
```diff
@@ -104,13 +112,15 @@ import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton.js'
...
 import NcIconSvgWrapper from '@nextcloud/vue/dist/Components/NcIconSvgWrapper.js'
+import NcUserBubble from '@nextcloud/vue/dist/Components/NcUserBubble.js'
 
 import FileInfo from '../services/FileInfo.js'
 import LegacyView from '../components/LegacyView.vue'
 import SidebarTab from '../components/SidebarTab.vue'
 import SystemTags from '../../../systemtags/src/components/SystemTags.vue'
 import logger from '../logger.ts'
 
+
```
Comment (susnux) [inline apps/files/src/views/Sidebar.vue:123]:
```suggestion
```

Comment (susnux) [reply apps/files/src/views/Sidebar.vue:123]: make eslint happy

File: apps/files/src/views/Sidebar.vue (hunk 6 above)
Comment (susnux) [reply apps/files/src/views/Sidebar.vue:494]: Yes I think the sidebar could be loaded before the files app, meaning Pinia is not yet created... A chick-egg problem we could solve but is unrelated here.
Would need to refactor all of the files stores...

**Review by susnux — APPROVED** [review]
We need to migrate the Sidebar to use proper Node data instead of legacy stuff, but until then this is probably the best way without patching legacy OC.Files.Client
